### PR TITLE
improve api docs

### PR DESCRIPTION
**Pull request by eordentlich**
- adapted sphinx set up from Pyspark for more similar look to api docs.
- sphinx logic to skip unsupported attributes and methods.
- avoid documenting non-user facing classes and apis

TBD - Additional work is required to:
- override Param attributes for which not all values are supported, e.g. solver for LinearRegression
- class constructor inputs should appear in documentation
- no unsupported methods (besides getters and setters for unsupported attributes which are handled in this PR) should appear in docs
- delete unsupported methods and attributes at runtime in code, or override with not implemented exceptions.
- document cuml versions of parameters 
- add possibly missing setters

### CODE REVIEW

**Comment by eordentlich** [comment]
build

File: python/src/spark_rapids_ml/utils.py [hunk 1]
```diff
@@ -28,6 +28,27 @@
...
 
 
+def _method_names_from_param(spark_param_name: str) -> List[str]:
+    cap = spark_param_name[0].upper() + spark_param_name[1:]
+    getter = f"get{cap}"
+    setter = f"set{cap}"
+    return [getter, setter]
+
+
+def _unsupported_methods_attributes(clazz: Any) -> Set[str]:
```
Comment (leewyang) [inline python/src/spark_rapids_ml/utils.py:38]: Probably should add pydoc for this function (and maybe also for above).

File: docs/source/spark_rapids_ml.rst [hunk 2]
```diff
@@ -1,93 +1,88 @@
...
+..  Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+
+Spark Rapids ML
+=======================
```
Comment (leewyang) [inline docs/source/spark_rapids_ml.rst:20]: Not sure if line length needs to match (but minimally, seems like convention)

**Review by leewyang — COMMENTED** [review]
Still need to review output of doc build.

**Comment by leewyang** [comment]
Is there any way we can remove the extra level (spark_rapids_ml -> Spark Rapids ML) in the main page?
Screenshot 2023-05-01 at 6 00 10 PM

**Comment by leewyang** [comment]
Upon spot check, it looks like the PCA docs now mostly match what you'd see in Jupyter auto-complete, which is good.  

However, the PCAModel docs looks like it might be missing some of the "cuml" attributes, e.g. `components_` and `explained_variance_ratio_`, which are actually wrapped by "spark" attributes like `pc` and `explainedVariance`.  That said, I think this is fine as-is.

**Review by leewyang — DISMISSED** [review]

**Comment by eordentlich** [comment]
Addressed comments.

Updating PR description for TBDs.

**Review by leewyang — APPROVED** [review]

**Comment by eordentlich** [comment]
build

**Pull request merged**